cdap-common/src/main/java/co/cask/cdap/common/io/CaseInsensitiveEnumTypeAdapterFactory.java

```diff
@@ -35,22 +35,37 @@
  * instead of deserializing lowercase only.
  */
 public class CaseInsensitiveEnumTypeAdapterFactory implements TypeAdapterFactory {
+
+  private final boolean useUpperCase;
+
+  public CaseInsensitiveEnumTypeAdapterFactory(boolean useUpperCase) {
```

Comment:
you could have a no-arg constructor that calls `this(false)`. That way you wouldn't have to change any of the other classes.